### ISSUE TITLE
Return statement does not break out of while-loop

### ISSUE THREAD

**Issue by ddjoh** [actonlang/acton#1194]
### Acton Version

0.14.2.20230221.16.25.49

### Steps to Reproduce

```
actor main(env):
    def foo():
        while True:
            print("SPAM")
            return
    foo()
    await async env.exit(0)
```

### Expected Behavior

Program to terminate after printing "SPAM" only once.

### Actual Behavior

Program does not terminate

**Comment by plajjan** [comment]
@nordlander we had some control flow related bugs where we said we wanted to fix exceptions first. Exceptions are now fixed and I revisited this, but it is still not fixed. Can you take a look at this?

**Comment by nordlander** [comment]
Fixed by #1490.
